### DISCUSSION TITLE
How can I set a custom image for cross reference hover preview?

### DISCUSSION THREAD

**Discussion by rubak** [quarto-dev/quarto-cli#10669] [Q&A] [answered]
### Description

Is there a way to set a custom image as the preview when you hover over a cross reference? When you have interactive content which isn't an image then the cross reference preview just shows a blank page. A reproducible example in R is below. The rendered version [is here](https://people.math.aau.dk/~rubak/quarto-hover.html) in case you need it:

````qmd
---
title: "quarto-hover"
format:
    html: 
      self-contained: true
---

Here is an simple R plot which will be displayed in the hover preview.
```{r}
#| label: fig-base-plot
plot(1:10)
```

Here is an interactive plot which won't be displayed in the hover preview.
```{r}
#| label: fig-plotly
#| message: false
library(plotly)
plot_ly(x = 1:10, y = 1:10, z = 10:1, type = 'scatter3d', mode = 'markers')
```

When you hover over the link to @fig-base-plot you see a preview, but not for @fig-plotly. 

````

**Comment by mcanouil** [comment]
Not directly in Quarto, but since it's HTML, you could probably use JavaScript.

Side note: `self-contained` is a 2-years old outdated option. You should use `embed-resources`.

**Reply by rubak** [reply]
Thanks a lot for the quick answer. Can you give any more hints as what I should do to fix this in HTML/JavaScript? How do I find the relevant HTML/JavaScript which is generating the hover effect so that I can modify it? I'm completely blank so any idea as to how I should attack the problem would be much appreciated.

**Reply by mcanouil** [reply, marked as answer]
Sorry, I can't and that is out of Quarto scope.
For advanced customisation, you are a bit on your own I'm afraid, as we cannot teach you HTML/JavaScript or LaTeX or Typst or OpenXML.

The library used is Tippy.js:

https://github.com/quarto-dev/quarto-cli/blob/3cfdf0178848bce308be22c0fad0ca420add762c/src/resources/formats/html/tippy/

**Reply by rubak** [reply]
Thanks a lot for pointing me in that direction. It is absolutely understandable that you cannot support manual tweaks to HTML/JavaScript. 

Is there a place to make a feature request for customizable cross reference previews, just so you have a record of the potential use case? Alternatively the request could be for turning previews off for individual cross references or specific classes of references to avoid blank previews for popular graphics such as 3D plotly graphs.

**Reply by mcanouil** [reply]
You could either edit this thread and change the category or open a new GitHub Discussion directly in the "feature request category.

Though, your request is a bit of niche and quarto team's plate is quite full, so very unlikely to happen in a near future.
